sapi_demo/src/EventSubscriber/EventViewEntitySubscriber.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+namespace Drupal\sapi_demo\EventSubscriber;
+
+use Drupal\Component\Plugin\PluginManagerInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\EventDispatcher\Event;
+use Drupal\Core\Session\AccountProxy;
+use Drupal\sapi\Dispatcher;
+use Drupal\Core\Routing\CurrentRouteMatch;
+use Symfony\Component\HttpKernel\KernelEvents;
+use Drupal\sapi\ActionTypeInterface;
+
+use Drupal\Core\Entity;
+
+/**
+ * Class EventViewEntitySubscriber.
+ *
+ * @package Drupal\sapi_demo
+ */
+class EventViewEntitySubscriber implements EventSubscriberInterface {
+
+  /**
+   * Drupal\Core\Session\AccountProxy definition.
+   *
+   * @var \Drupal\Core\Session\AccountProxy
+   */
+  protected $currentUser;
+
+  /**
+   * Drupal\sapi\Dispatcher definition.
+   *
+   * @var \Drupal\sapi\Dispatcher
+   */
+  protected $sapiDispatcher;
+
+  /**
+   * The statistics action type plugin manager which will be used to create sapi
+   * items to be passed to the dispatcher
+   *
+   * @var \Drupal\Component\Plugin\PluginManagerInterface $sapiActionTypeManager
+   */
+  protected $sapiActionTypeManager;
+
+  /**
+   * Drupal\Core\Routing\CurrentRouteMatch definition.
+   *
+   * @var \Drupal\Core\Routing\CurrentRouteMatch
+   */
+  protected $currentRouteMatch;
+
+  /**
+   * Constructor.
+   */
+  public function __construct(AccountProxy $currentUser, Dispatcher $sapiDispatcher, PluginManagerInterface $sapiActionTypeManager, CurrentRouteMatch $currentRouteMatch) {
+    $this->currentUser = $currentUser;
+    $this->sapiDispatcher = $sapiDispatcher;
+    $this->sapiActionTypeManager = $sapiActionTypeManager;
+    $this->currentRouteMatch = $currentRouteMatch;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  static function getSubscribedEvents() {
+    // Priority is set to 1 to avoid this listener from being stopped by other
+    // listeners.
+    // @see ContainerAwareEventDispatcher::dispatch()
+    $events[KernelEvents::VIEW][] = ['onEventView', 1];
```

Comment:
No change needed, just a note: REST request to `/node/1?_format=json` will not trigger `KernelEvents::VIEW` because `rest` module returns `Symfony\Component\HttpFoundation\Response` (through a child class). If REST view requests are needed, a separate listener on `KernelEvents::RESPONSE` needs to be added as a separate method.